src/main/java/seedu/address/model/person/ModuleRoleMap.java

```diff
@@ -177,4 +178,21 @@ public List<ModuleRolePair> getData() {
                 .map(entry -> new ModuleRolePair(entry.getKey(), entry.getValue()))
                 .collect(Collectors.toList());
     }
+
+    /**
+     * Converts the ModuleRoleMap back to a list of module-role pairs.
+     *
+     * @return a list of module-role pairs.
+     */
+    public List<String> toModuleRolePairs() {
+        List<String> moduleRolePairs = new ArrayList<>();
+
+        for (Map.Entry<ModuleCode, RoleType> entry : roles.entrySet()) {
+            ModuleCode moduleCode = entry.getKey();
+            RoleType roleType = entry.getValue();
+            moduleRolePairs.add(moduleCode + "-" + roleType);
+        }
```

Comment:
Perhaps we can make use of the toString method in ModuleRolePair? maybe we can convert it to List<ModuleRolePair> using getData() then invoke the toString method and make it a list to make it looks like being written by a same person (?)